Modelica/Resources/BuildProjects/CMake/CMakeLists.txt

```
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.15)
+
+if(POLICY CMP0091)
+  cmake_policy(SET CMP0091 NEW)
+  if(WIN32)
```

```suggestion
  if(WIN32 AND NOT MINGW)
```

I don't have MINGW available to test, but maybe like this then?

---

Actually, since this is for `MSVC`, maybe it should just be `if(MSVC)`?

---

> Actually, since this is for `MSVC`, maybe it should just be `if(MSVC)`?

Probably, yes.

---

> I don't have MINGW available to test, but maybe like this then?

It's tested in CI checks.

---

From the CI check log, I assume the `MINGW` variable is not set that early, which I guess shouldn't surprise that much. The documentation says:

> Set to a true value when at least one language is enabled with a compiler targeting the GNU ABI on Windows (MinGW).

which I'm guessing happens later on in the CMake process.

But we need to do this very early like we are doing it now, or the `CMAKE_MSVC_RUNTIME_LIBRARY` variable won't have any effect.

I think we should just stick with `if(WIN32)` then. The setting of `CMAKE_MSVC_RUNTIME_LIBRARY` should not hurt `MinGW`, since it is MSVC-specific. We could reformulate the message, or not have a message at all, if we want to avoid the confusing log output.